### PR TITLE
429 alert overutilization

### LINKED ISSUE #429: Overutilization alert during enrollment
**AC:**
- If a provider has filled all slots for a given age group in the current reporting period, display an alert letting the user know they will have used more spaces than are available.
- The alert logic is based on spaces allotted per age group (e.g., if the site has 12 preschool spaces, this alert shows if there are currently 12 children enrolled in a preschool space while a user tries to enroll a new child in a preschool space)
- Copy (note differs from design): 
  - This space type is already fully utilized for the [report period, e.g., January 2019] reporting period.

**Design:**
Screen Shot 2019-12-17 at 2 18 21 PM

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #429

### CODE REVIEW

File: src/Hedwig/ClientApp/src/utils/models/reportingPeriod.ts [hunk 1]
```diff
@@ -20,7 +20,8 @@ export const currentReportingPeriod = (periods: ReportingPeriod[]): ReportingPer
 	const now = moment();
 	const startOfMonthDate = moment(`${now.format('YYYY-MM')}-01`);
 	// Assumes that ReportingPeriod.Period will always be on the first day of the month
-	return periods.find(period => period.period.getTime() === startOfMonthDate.valueOf());
+	// Compare with dates, not times to avoid timezone ridiculousness
+	return periods.find(period => moment(period.period).format('YYYY-MM-DD') === startOfMonthDate.format('YYYY-MM-DD'));
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/utils/models/reportingPeriod.ts]: can leverage `moment.date` as opposed to using format

File: src/Hedwig/Repositories/HedwigRepository.cs [hunk 2]
```diff
@@ -34,9 +34,8 @@ public Task SaveChangesAsync()
 		/// <param name="updates"></param>
 		/// <param name="currents"></param>
 		/// <typeparam name="T"></typeparam>
-		public void UpdateEnumerableChildObjects<T>(IEnumerable<IHedwigIdEntity<T>> updates, IEnumerable<IHedwigIdEntity<T>> currents)
+		public void UpdateEnumerableChildObjects<T>(IEnumerable<IHedwigIdEntity<T>> updates, IEnumerable<IHedwigIdEntity<T>> currents) where T : IEquatable<T>
```
Comment (jlhogan) [inline src/Hedwig/Repositories/HedwigRepository.cs]: line 46 changed to
```
if(!updates.Any(u => u.Id.Equals(current.Id)))
```
which may mean the `where T : IEquatable<T>` is unnecessary

File: src/Hedwig/ClientApp/src/utils/fundingSelectionUtils.ts [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
-		FundingSelection.PRIVATE_PAY :
-		value === '' ?
-			FundingSelection.UNSELECTED :
-			fundingTimeFromString(value) === FundingTime.Full ?
-				FundingSelection.CDC_FULL :
-				FundingSelection.CDC_PART;
-}
\ No newline at end of file
+	const source =
+		value === 'privatePay'
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/utils/fundingSelectionUtils.ts]: I find the staggered indented on nested ternarys extremely helpful in readability. Was this an auto change or manual? if auto, can we update a linting rule to leave these be? Not a blocker for this PR, just my opinion

Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/utils/fundingSelectionUtils.ts]: I second the value of nested formatting here!

Comment (mmazanec22) [reply src/Hedwig/ClientApp/src/utils/fundingSelectionUtils.ts]: I agree that it's more readable; this was an auto update.  I will check and see if there's a linter rule that will preserve it!

File: src/Hedwig/ClientApp/src/utils/models/reportingPeriod.ts (hunk 1 above)
Comment (mmazanec22) [reply src/Hedwig/ClientApp/src/utils/models/reportingPeriod.ts]: I think that particular method will not be what we want here: https://momentjs.com/docs/#/get-set/date/

**Pull request merged**